File: scripts/task_tested_with_compile_script/function-composition/perl/function-composition.pl
sub compose {
    my ($f, $g) = @_;

    sub {
        $f -> ($g -> (@_))
    };
}

use Math::Trig;

my $r = 1;

sub executeTask {
	my ($i) = @_;
   	return compose(sub {sin $_[0]}, \&asin)->($i);
}

for($a = 0; $a < 1000000000; $a += 1) { 
	$r = executeTask($a);
}

File: scripts/task_tested_with_compile_script/function-composition/r/function-composition.r


executeTask <- function(i) {

    	compose <- function(f,g) function(x) { f(g(x)) }    
    	t <- compose(sin, asin)
    	#t(i)
	return(t(i))
}

r = 1

for (i in 0:1000000000) {
	r = executeTask(r + i)
}
